pkg/clients/database/sql.go

```diff
@@ -0,0 +1,612 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package database
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/services/mysql/mgmt/2017-12-01/mysql"
+	"github.com/Azure/azure-sdk-for-go/services/mysql/mgmt/2017-12-01/mysql/mysqlapi"
+	"github.com/Azure/azure-sdk-for-go/services/postgresql/mgmt/2017-12-01/postgresql"
+	"github.com/Azure/azure-sdk-for-go/services/postgresql/mgmt/2017-12-01/postgresql/postgresqlapi"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/pkg/errors"
+
+	"github.com/crossplaneio/crossplane-runtime/pkg/meta"
+
+	azuredbv1alpha3 "github.com/crossplaneio/stack-azure/apis/database/v1alpha3"
+	azure "github.com/crossplaneio/stack-azure/pkg/clients"
+)
+
+// State strings for MySQL and PostgreSQL.
+const (
+	StateDisabled = string(mysql.ServerStateDisabled)
+	StateDropping = string(mysql.ServerStateDropping)
+	StateReady    = string(mysql.ServerStateReady)
+)
+
+var (
+	skuShortTiers = map[mysql.SkuTier]string{
+		mysql.Basic:           "B",
+		mysql.GeneralPurpose:  "GP",
+		mysql.MemoryOptimized: "MO",
+	}
+)
+
+// MySQLServerAPI represents the API interface for a MySQL Server client
+type MySQLServerAPI interface {
+	ServerNameTaken(ctx context.Context, s *azuredbv1alpha3.MySQLServer) (bool, error)
+	GetServer(ctx context.Context, s *azuredbv1alpha3.MySQLServer) (mysql.Server, error)
+	CreateServer(ctx context.Context, s *azuredbv1alpha3.MySQLServer, adminPassword string) error
+	UpdateServer(ctx context.Context, s *azuredbv1alpha3.MySQLServer) error
+	DeleteServer(ctx context.Context, s *azuredbv1alpha3.MySQLServer) error
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+// MySQLServerClient
```

Comment:
I'd personally be happy to see these "demarcation" comment headers go away - I think they're unique to some of the older Azure codebase. If we think this file is long and complex enough that we need comments like these to make it more navigable we might consider instead breaking it up into multiple files.